### PR TITLE
480 single record batch edit

### LINKED ISSUE #480: Single-record "batch edit"
Blocked by #474 

**AC**:
- using same form-creation logic from batch edit, enable user to edit a single record in a steplist, where sections are only visible if they have field with needed information and fields are only visible if they need information
- on completion of steplist, user is shown success info and back to roster button
- clicking on missing info icon brings user to this view

## Designs
9  Single Child Add needed info

batch-edit-single

### LINKED ISSUE #566: Display content for batch edit: Show surrounding data fields to give context.
Content to show in groups or alone:
- First name, Middle Name, Last Name, Suffix
- Birth date
- Birth Certificate
- Race and Ethnicity
- Gender
- Receiving disability services
- Dual Language Learner
- Child lives with foster family
- Address
    - Street address 
    - Town 
    - State
    - Zip code
- Homelessness status
- Family income determination
    - Household size
    - Income
    - Determination date
- Enrollment and funding
    - Site, Model Type, Age group, enrollment date
    - Funding source, space type, first reporting period

### PR DESCRIPTION
**Pull request by jlhogan**
Closes #480 
closes #566 
TODO: fixup tests, add tests

### CODE REVIEW

**Review by mmazanec22 — APPROVED** [review]

File: client/src/shared/models/db/Child.ts [hunk 1]
```diff
@@ -5,6 +5,14 @@ import { Moment } from 'moment';
 import { ObjectWithValidationErrors } from '../ObjectWithValidationErrors';
 import { BirthCertificateType } from '../BirthCertificateType';
 
+export const RACE_FIELDS = [
```
Comment (JordanGuinn) [inline client/src/shared/models/db/Child.ts:8]: I think this already exists in `src/utils/raceFields.ts`?

Comment (jlhogan) [reply client/src/shared/models/db/Child.ts:8]: oh i replaced the use of that with this but then maybe didn't clean itup there?

**Review by JordanGuinn — APPROVED** [review]

File: client/src/shared/models/db/Child.ts (hunk 1 above)
Comment (JordanGuinn) [reply client/src/shared/models/db/Child.ts:8]: Yeah - these fields are still present there as well 😕

Comment (jlhogan) [reply client/src/shared/models/db/Child.ts:8]: that is now deleted !

**Comment by RyanHansz** [comment]
@lucasolivia Can you take a look at this as well?

Here's what I noticed, let me know if any of these are larger lifts:
- Single record on the roster - add a tooltip on hover that says “Add missing info”
- Navigating to a child record and back moves the record to the bottom of the list (Ideally it would stay in the same location)
- Add 1 em padding to below:
image
- Remove padding-top from:
image
- Update text for missing date of birth, age group and or site:
image
- Remove “errors in” from:
image
- When looking at a completed record in the list:
image

Change roster message from:
image
To:
image

Note to eventually add "alert with button" to component library

**Comment by RyanHansz** [comment]
@jlhogan can you look at these updates and let me know which ones are enough work to require their own tickets?

**Comment by jlhogan** [comment]
ah yes so sorry! completely forgot about this my bad

**Pull request merged**